configserver/src/main/java/com/yahoo/vespa/config/server/tenant/EndpointCertificateRetriever.java

```diff
@@ -0,0 +1,37 @@
+// Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.config.server.tenant;
+
+import com.yahoo.config.model.api.EndpointCertificateMetadata;
+import com.yahoo.config.model.api.EndpointCertificateSecrets;
+import com.yahoo.container.jdisc.secretstore.SecretStore;
+
+import java.util.Optional;
+
+/**
+ * Used to retrieve actual endpoint certificate/key from secret store.
+ *
+ * @author andreer
+ */
+public class EndpointCertificateRetriever {
+
+    private final SecretStore secretStore;
+
+    public EndpointCertificateRetriever(SecretStore secretStore) {
+        this.secretStore = secretStore;
+    }
+
+    public Optional<EndpointCertificateSecrets> readEndpointCertificateSecrets(EndpointCertificateMetadata metadata) {
+        return Optional.of(readFromSecretStore(metadata));
+    }
+
+    private EndpointCertificateSecrets readFromSecretStore(EndpointCertificateMetadata endpointCertificateMetadata) {
```

Comment:
We've had cases where the version of private key and cert were incompatible. Would it make sense to validate that the cert is indeed issued using the private key?

Reply:
Yes, I think I have a ticket on this too. Might as well add it now.